### PR TITLE
Phoenix/more compartment models

### PR DESCRIPTION
**Pull request by phoenixAja**
add more simple compartment models, for now the model list includes:
```
        choices=[
            "SimpleSIRModel",
            "SimpleSEIRModel",
            "SimpleSEIRDModel",
            "OverdispersedSIRModel",
            "OverdispersedSEIRModel",
            "SuperspreadingSIRModel",
            "SuperspreadingSEIRModel",
        ],
```
for examples on use see the Makefile and test csv files below for population and num_samples: 
https://raw.githubusercontent.com/czbiohub/test-datasets/epigen/testdata/epigen/num_samples.csv
population:
'https://raw.githubusercontent.com/czbiohub/test-datasets/epigen/testdata/epigen/population.csv'

### CODE REVIEW

**Comment by phoenixAja** [comment]
took `HeterogeneousSIRModel` out for now, getting a binomial error (see example here: https://github.com/czbiohub/nf-epigen/runs/1361214687)

File: Makefile [hunk 1]
```diff
@@ -6,13 +6,44 @@ GISAID_SEQUENCES=gisaid/sequences.tsv
...
+	nextflow run main.nf \
+	-profile docker \
+	-resume \
+	--newicks ${NEWICKS} \
+	--gisaid_metadata ${GISAID_METADATA} \
+	--gisaid_sequences ${GISAID_SEQUENCES} \
+	--model_type SimpleSEIRDModel \
+	--population_csv ${POPULATION_CSV} \
+	--num_samples_csv ${NUM_SAMPLES_CSV} \
+	--mortality_rate 2
```
Comment (lucymli) [inline Makefile:46]:
```suggestion
	--mortality_rate 0.01
```

Comment (lucymli) [reply Makefile:46]: Mortality rate has to be between 0 and 1.

**Review by lucymli — COMMENTED** [review]
It would be helpful to save the output of different models in different results folders. E.g. simpleseirdmodel in results/simpleseird, superspreadingseirmodel in results/superspreadingseir.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,7 +17,12 @@ jobs:
         # Nextflow versions: check pipeline minimum and current latest
         nxf_ver: ['19.10.0', '']
         profile_flags: [
-        'test',
+        'test_overdispersedseirmodel',
```
Comment (lucymli) [inline .github/workflows/ci.yml:20]: When I tested using `nextflow run main.nf -profile docker,test_overdispersedseirmodel --newicks 'timetrees/*.nwk'` I got this error: Missing output file(s) `metadata.tsv` expected by process `impute_infection_dates (unitedkingdom)`

Comment (lucymli) [reply .github/workflows/ci.yml:20]: It worked for all the previous 10 locations but error when it reached unitedkingdom

Comment (phoenixAja) [reply .github/workflows/ci.yml:20]: i haven't tried using the config files and then specifying the `--newicks` flag separately. Let me try it on my side, maybe it's having a hard time downloading the metadata file

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -17,7 +17,12 @@ jobs:
...
+        'test_overdispersedsirmodel',
```
Comment (lucymli) [inline .github/workflows/ci.yml:21]: Similarly for this model, I got this error for Japan: `Missing output file(s) 'metadata.tsv' expected by process 'impute_infection_dates (japan)'`

Comment (phoenixAja) [reply .github/workflows/ci.yml:21]: that's wierd because the config file tests were passing on github actions.. let me investigate a bit

**Comment by lucymli** [comment]
All the other models worked except the two I highlighted in the comments.

**Comment by phoenixAja** [comment]
@lucymli i'm testing for those two models now, were you using the same regions when testing each of the different models?

**Comment by lucymli** [comment]
> @lucymli i'm testing for those two models now, were you using the same regions when testing each of the different models?

Yes I tested with all 12 regions

**Comment by phoenixAja** [comment]
> > @lucymli i'm testing for those two models now, were you using the same regions when testing each of the different models?
> 
> Yes I tested with all 12 regions

were the models failing with just `unitedkingdom` and `japan`? I just tested the overdispersedSEIRmodel with `california` and `minnesota` timetrees and it worked for me. Maybe there was a blip in your internet connection when downloading the metadata file (in which case retrying would fix the issue) otherwise maybe something weird is going on with the `japan` and `unitedkingdom` timetrees.
```

(base) phoenix@tesla:~/code/nf-epigen$ nextflow run main.nf -profile docker,test_overdispersedseirmodel --newicks 'timetrees/*.nwk'
N E X T F L O W  ~  version 20.04.1
Launching `main.nf` [hungry_lichterman] - revision: 2c8730c303
----------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
  nf-core/epigen v1.0dev
----------------------------------------------------
Run Name          : hungry_lichterman
Max Resources     : 6 GB memory, 2 cpus, 2d time per job
Container         : docker - czbiohub/epigen:latest
Output dir        : ./results
Launch dir        : /home/phoenix/code/nf-epigen
Working dir       : /home/phoenix/code/nf-epigen/work
Script dir        : /home/phoenix/code/nf-epigen
User              : phoenix
Config Profile    : docker,test_overdispersedseirmodel
Config Description: Minimal test dataset to check pipeline function
----------------------------------------------------
executor >  local (8)
[9b/e6b207] process > download_timeseries                 [100%] 1 of 1 ✔
[e0/3b0d6c] process > clean_and_transform_timeseries (1)  [100%] 1 of 1 ✔
[32/b6f136] process > impute_infection_dates (california) [100%] 2 of 2 ✔
[7e/b0b0c4] process > seir_model (2)                      [100%] 2 of 2 ✔
[0b/d49a0d] process > get_software_versions               [100%] 1 of 1 ✔
[33/8fc720] process > output_documentation                [100%] 1 of 1 ✔
[0;35m[nf-core/epigen] Pipeline completed successfully
WARN: To render the execution DAG in the required format it is required to install Graphviz -- See http://www.graphviz.org for more info.
Completed at: 06-Nov-2020 11:28:24
Duration    : 3m 40s
CPU hours   : 0.1
Succeeded   : 8
```

**Comment by lucymli** [comment]
> > > @lucymli i'm testing for those two models now, were you using the same regions when testing each of the different models?
> > 
> > 
> > Yes I tested with all 12 regions
> 
> were the models failing with just `unitedkingdom` and `japan`? I just tested the overdispersedSEIRmodel with `california` and `minnesota` timetrees and it worked for me. Maybe there was a blip in your internet connection when downloading the metadata file (in which case retrying would fix the issue) otherwise maybe something weird is going on with the `japan` and `unitedkingdom` timetrees.
> 
> ```
> 
> (base) phoenix@tesla:~/code/nf-epigen$ nextflow run main.nf -profile docker,test_overdispersedseirmodel --newicks 'timetrees/*.nwk'
> N E X T F L O W  ~  version 20.04.1
> Launching `main.nf` [hungry_lichterman] - revision: 2c8730c303
> ----------------------------------------------------
>                                         ,--./,-.
>         ___     __   __   __   ___     /,-._.--~'
>   |\ | |__  __ /  ` /  \ |__) |__         }  {
>   | \| |       \__, \__/ |  \ |___     \`-._,-`-,
>                                         `._,._,'
>   nf-core/epigen v1.0dev
> ----------------------------------------------------
> Run Name          : hungry_lichterman
> Max Resources     : 6 GB memory, 2 cpus, 2d time per job
> Container         : docker - czbiohub/epigen:latest
> Output dir        : ./results
> Launch dir        : /home/phoenix/code/nf-epigen
> Working dir       : /home/phoenix/code/nf-epigen/work
> Script dir        : /home/phoenix/code/nf-epigen
> User              : phoenix
> Config Profile    : docker,test_overdispersedseirmodel
> Config Description: Minimal test dataset to check pipeline function
> ----------------------------------------------------
> executor >  local (8)
> [9b/e6b207] process > download_timeseries                 [100%] 1 of 1 ✔
> [e0/3b0d6c] process > clean_and_transform_timeseries (1)  [100%] 1 of 1 ✔
> [32/b6f136] process > impute_infection_dates (california) [100%] 2 of 2 ✔
> [7e/b0b0c4] process > seir_model (2)                      [100%] 2 of 2 ✔
> [0b/d49a0d] process > get_software_versions               [100%] 1 of 1 ✔
> [33/8fc720] process > output_documentation                [100%] 1 of 1 ✔
> [0;35m[nf-core/epigen] Pipeline completed successfully
> WARN: To render the execution DAG in the required format it is required to install Graphviz -- See http://www.graphviz.org for more info.
> Completed at: 06-Nov-2020 11:28:24
> Duration    : 3m 40s
> CPU hours   : 0.1
> Succeeded   : 8
> ```

Yes it only blipped for those two regions which was weird. I'll try again.

**Comment by phoenixAja** [comment]
> It would be helpful to save the output of different models in different results folders. E.g. simpleseirdmodel in results/simpleseird, superspreadingseirmodel in results/superspreadingseir.

good idea! just pushed a change to do this 👍

**Comment by phoenixAja** [comment]
@lucymli i was able to run all 12 locations with `OverdispersedSEIRModel` using the `Makefile` (`make run-nf-epigen`), all the data to do the same run that i did is located on `data_lg` so you should be able to run that rule just fine. The pipeline took about 5 mins to finish on `tesla`. I think the test profile for `overdispersedseir` was failing because the test config profiles all use a shortened version of the `gisaid_metadata` and `gisaid_sequences`.

**Comment by lucymli** [comment]
Your makefile worked for me!

**Pull request merged**